frontend/components/TableContainer/utilities/config_utils.ts

```diff
@@ -0,0 +1,56 @@
+// Note: This is a lite solution and does NOT include ability to
+// "Select all across pages" nor to disable the header checkbox
+
+import { HeaderProps, Row } from "react-table";
+
+interface GetConditionalSelectHeaderCheckboxProps {
+  /** react-table header props */
+  headerProps: React.PropsWithChildren<HeaderProps<any>>;
+  /** A function defining which rows are selectable */
+  checkIfRowIsSelectable: (row: Row<any>) => boolean;
+}
+
+const getConditionalSelectHeaderCheckboxProps = ({
```

Comment:
It seems like there's a lot of iterating over all rows going on here. Not necessarily a problem, but we should keep an eye out for potential performance impacts.

Reply:
Yeah, luckily for policies it's only 20 rows, but I know queries is client side pagination so that could be a lot of iterating. Going to comb through Jacob's queries refactor after this and see if he took that into account!